packages/synapse-react-client/src/components/Webhook/WebhookDashboard.test.tsx

```diff
@@ -0,0 +1,267 @@
+
+    // Create a webhook
+    await user.click(createWebhookButton)
+    await createWebhook(user)
+
+    await waitFor(() => {
+      expect(screen.queryByRole('dialog')).not.toBeInTheDocument()
+      expect(screen.getAllByRole('row')).toHaveLength(2)
+    })
+
+    // Verify the webhook
+    const verifyCodeDialogButton = await submitVerificationCodeForWebhook(user)
+
+    await waitFor(() => {
+      expect(verifyCodeDialogButton).not.toBeInTheDocument()
+      expect(screen.getByText('Verified')).toBeInTheDocument()
+    })
+
+    // Update the webhook
+    await user.click(screen.getByRole('button', { name: 'Edit' }))
+    const editWebhookDialog = await screen.findByRole('dialog')
+    // Verify the 'Danger Zone' appears
+    within(editWebhookDialog).getByText(/danger zone/i)
+    await user.click(
+      within(editWebhookDialog).getByRole('checkbox', { name: 'Update' }),
+    )
+    const updateButton = within(editWebhookDialog).getByRole('button', {
+      name: 'Update Webhook',
+    })
+    await user.click(updateButton)
+
+    await waitFor(() => {
+      expect(editWebhookDialog).not.toBeInTheDocument()
+    })
+
+    // Delete the webhook
+    await user.click(screen.getByRole('button', { name: 'Edit' }))
+    const deleteButton = await screen.findByRole('button', {
+      name: 'Delete Webhook',
+    })
+    await user.click(deleteButton)
+    const deleteDialog = await screen.findByRole('dialog', {
+      name: 'Confirm Delete Webhook?',
+    })
+    const confirmDeleteButton = within(deleteDialog).getByRole('button', {
+      name: 'Delete',
+    })
+    await user.click(confirmDeleteButton)
+
+    await waitFor(() => {
+      expect(screen.queryByRole('dialog')).not.toBeInTheDocument()
+      screen.getByText(/You have not created any webhooks/)
+      expect(screen.queryByRole('row')).not.toBeInTheDocument()
+    })
+  })
+
+  it('Allows requesting a new verification code when the verification status is FAILED', async () => {
+    // Preload the mock service to return a webhook with verification status FAILED
+    webhookService.clear()
+    webhookService.create(
+      WebhookToJSON({
+        objectId: mockProject.id,
+        objectType: SynapseObjectType.ENTITY,
+        eventTypes: new Set(['CREATE']),
+        verificationStatus: WebhookVerificationStatus.FAILED,
+      }),
+    )
+
+    const sendVerificationCodeSpy = jest.spyOn(
+      MOCK_CONTEXT_VALUE.synapseClient.webhookServicesClient,
+      'putRepoV1WebhookWebhookIdVerificationCode',
+    )
+
+    const { user } = await renderComponent()
+    await screen.findByText(/Failed/)
+    const resendCodeButton = await screen.findByRole('button', {
+      name: 'Resend code',
+    })
+    await user.click(resendCodeButton)
+
+    await waitFor(() => {
+      expect(sendVerificationCodeSpy).toHaveBeenCalled()
+    })
+  })
+
+  it('Shows an error when verification fails due to code expiration', async () => {
```

Comment:
Tricky case, if your webhook code is invalid, the server returns 200 with `"isValid": false` in the body